src/main/java/org/opensearch/neuralsearch/processor/chunker/FixedTokenLengthChunker.java

```diff
@@ -0,0 +1,159 @@
+     * tokenizer should be string
+     */
+    @Override
+    public void validateAndParseParameters(Map<String, Object> parameters) {
+        this.tokenLimit = validatePositiveIntegerParameter(parameters, TOKEN_LIMIT_FIELD, DEFAULT_TOKEN_LIMIT);
+        this.overlapRate = validateRangeDoubleParameter(
+            parameters,
+            OVERLAP_RATE_FIELD,
+            OVERLAP_RATE_LOWER_BOUND,
+            OVERLAP_RATE_UPPER_BOUND,
+            DEFAULT_OVERLAP_RATE
+        );
+        this.tokenizer = validateStringParameters(parameters, TOKENIZER_FIELD, DEFAULT_TOKENIZER, false);
+        if (!WORD_TOKENIZERS.contains(this.tokenizer)) {
+            throw new IllegalArgumentException(
+                String.format(
+                    Locale.ROOT,
+                    "tokenizer [%s] is not supported for [%s] algorithm. Supported tokenizers are %s",
+                    this.tokenizer,
+                    ALGORITHM_NAME,
+                    WORD_TOKENIZERS
+                )
+            );
+        }
+    }
+
+    /**
+     * Return the chunked passages for fixed token length algorithm
+     *
+     * @param content input string
+     * @param runtimeParameters a map for runtime parameters, containing the following runtime parameters:
+     * 1. max_token_count the max token limit for the tokenizer
+     */
+    @Override
+    public List<String> chunk(String content, Map<String, Object> runtimeParameters) {
+        // before chunking, validate and parse runtimeParameters
+        int maxTokenCount = validatePositiveIntegerParameter(runtimeParameters, MAX_TOKEN_COUNT_FIELD, DEFAULT_MAX_TOKEN_COUNT);
+
+        List<AnalyzeToken> tokens = tokenize(content, tokenizer, maxTokenCount);
+        List<String> chunkResult = new ArrayList<>();
+
+        int startTokenIndex = 0;
+        int startContentPosition, endContentPosition;
+        int overlapTokenNumber = (int) Math.floor(tokenLimit * overlapRate);
+
+        while (startTokenIndex < tokens.size()) {
+            if (startTokenIndex == 0) {
+                // include all characters till the start if no previous passage
+                startContentPosition = 0;
+            } else {
+                startContentPosition = tokens.get(startTokenIndex).getStartOffset();
+            }
+            if (startTokenIndex + tokenLimit >= tokens.size()) {
+                // include all characters till the end if no next passage
+                endContentPosition = content.length();
+                chunkResult.add(content.substring(startContentPosition, endContentPosition));
+                break;
+            } else {
+                // include gap characters between two passages
+                endContentPosition = tokens.get(startTokenIndex + tokenLimit).getStartOffset();
+                chunkResult.add(content.substring(startContentPosition, endContentPosition));
+            }
+            startTokenIndex += tokenLimit - overlapTokenNumber;
```

Comment:
is this guaranteed to increase with each call? trying to avoid infinite loop here

Reply:
Yes, the overlap rate is within [0, 0.5]. And we take math floor to obtain the overlap token number.
`overlapTokenNumber = (int) Math.floor(tokenLimit * overlapRate);`
Even when the overlap rate is 0.5 and token limit is 1, the overlap token number is 0.